src/codec.rs

```diff
@@ -609,12 +609,12 @@ impl Decode for Compact<u8> {
 			0 => prefix as u8 >> 2,
 			1 => {
 				let x = u16::decode(&mut PrefixInput{prefix: Some(prefix), input})? >> 2;
-				if x < 256 {
+				if x > 0b00111111 && x < 255 + 1 {
```

Comment:
Why always + 1 and not just `<=`?

Reply:
:) that is an old habit, most likely totally useless now ( https://stackoverflow.com/questions/1029992/is-the-inequality-operator-faster-than-the-equality-operator ).

Reply:
Okay, I remember that `> 0` is fast than `0 <`. Or something similar ^^